### PR TITLE
Correct token count for GPT-4v images

### PR DESCRIPTION
**Pull request by BeibinLi**
## Checks

- [X] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [X] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [X] I've made sure all auto checks have passed.

### CODE REVIEW

File: test/test_token_count.py [hunk 1]
```diff
@@ -83,7 +90,38 @@ def test_model_aliases():
     assert get_max_token_limit("gpt4-32k") == get_max_token_limit("gpt-4-32k")
 
 
+@pytest.mark.skipif(not img_util_imported, reason="img_utils not imported")
+def test_num_tokens_from_gpt_image():
```
Comment (sonichi) [inline test/test_token_count.py:94]: Is this test covered by CI?

File: test/agentchat/contrib/test_img_utils.py [hunk 2]
```diff
@@ -290,5 +291,24 @@ def test_formatting(self):
         self.assertEqual(result, expected_output)
 
 
+@pytest.mark.skipif(skip, reason="dependency is not installed")
```
Comment (sonichi) [inline test/agentchat/contrib/test_img_utils.py:294]: Is this test covered by CI?

**Comment by afourney** [comment]
Looks good, but have you tested these numbers with the OpenAI token calculator they publish? I'm imagining there are some subtleties to confirm (e.g., any tile overlap?)

File: autogen/token_count_utils.py [hunk 3]
```diff
@@ -125,7 +134,25 @@ def _num_token_from_messages(messages: Union[List, Dict], model="gpt-3.5-turbo-0
...
+                    if part["type"] == "text":
+                        num_tokens += len(encoding.encode(part["text"]))
+                    if "image_url" in part:
+                        assert "url" in part["image_url"]
+                        if not img_util_imported:
+                            warnings.warn(
+                                "img_utils or PIL not imported. Skipping image token count."
+                                "Please install autogen with [lmm] option.",
+                                ImportWarning,
+                            )
```
Comment (WaelKarkoub) [inline autogen/token_count_utils.py:151]: If the context history has 100 images, will this be printed 100 times?

**Review by WaelKarkoub — COMMENTED** [review]
Changes make sense to me. I just had a question regarding supporting other models (gemini, claude, etc...) Are we saying gpt4v is a good estimate for all models or will there be a plan to support other models as well?

File: autogen/token_count_utils.py (hunk 3 above)
Comment (afourney) [reply autogen/token_count_utils.py:151]: Good news! Only 10 images are supported by OpenAI  :) But yes... good question!

**Comment by julie2025** [comment]
Hi everyone,
I'm having an issue with the token count in MultimodalConversableAgent. It seems to be including the base64 encoded image URL when estimating the number of tokens and cost, which is incorrect.
Does anyone have any suggestions on how to fix this?

image

image

Thanks in advance.

**Pull request closed without merging**